sale_stock_release_channel_partner_by_date/models/sale_order.py

```diff
@@ -42,7 +42,8 @@ def _compute_release_channel_id(self):
             if not rec._check_release_channel_partner_date_requirements():
                 continue
             channel_date = rec.release_channel_partner_date_id
-            rec.release_channel_id = channel_date.release_channel_id
+            if channel_date:
+                rec.release_channel_id = channel_date.release_channel_id
```

Comment:
```suggestion
            forced_release_channel_id = self.env.context.get("order_release_channel_id", {}).get(rec.id)
            if not channel_date and forced_release_channel_id:
                rec.release_channel_id = forced_release_channel_id
                continue
            rec.release_channel_id = channel_date.release_channel_id
```

Reply:
That implementation does seem to be possible:
1. I don't know why but context variable passed in create order_release_channel_id never reach `_compute_release_channel_id`
2. We can't know the id of the record before creating it.